inception/inception-search-mtas/src/main/java/de/tudarmstadt/ukp/inception/search/index/mtas/MtasDocumentIndex.java

```diff
@@ -396,6 +402,221 @@ public long numberOfQueryResults(SearchQueryRequest aRequest)
+                String searchString = "<" + layer.getUiName().replace(' ', '_') + "."
+                        + feature.getUiName().replace(' ', '_') + "=\"\"/>";
+
+                Map<String, Double> results = getLayerStatistics(aStatisticRequest, searchString,
+                        fullDocSet, false);
+                results.remove("stats");
+                results.remove("sourceNumber");
+                results.put("Number of Documents", results.get("n"));
+                results.remove("n");
+                allStats.put(layer.getUiName() + "." + feature.getUiName(), results);
+                if (results.get("max") > 0.0) {
+                    nonNullStats.put(layer.getUiName() + "." + feature.getUiName(), results);
+                }
+
+                results = getLayerStatistics(aStatisticRequest, searchString, fullDocSet, true);
+                results.remove("stats");
+                results.remove("sourceNumber");
+                results.put("Number of Documents", results.get("n"));
+                results.remove("n");
+                allStats.put("per Sentence: " + layer.getUiName() + "." + feature.getUiName(),
+                        results);
+                if (results.get("max") > 0.0) {
+                    nonNullStats.put(
+                            "per Sentence: " + layer.getUiName() + "." + feature.getUiName(),
+                            results);
+                }
+            }
+        }
+        Map<String, Double> results = getLayerStatistics(aStatisticRequest, "<Token=\"\"/>",
+                fullDocSet, false);
+        results.remove("stats");
+        results.remove("sourceNumber");
+        results.put("Number of Documents", results.get("n"));
+        results.remove("n");
+        allStats.put("Token Count", results);
+        nonNullStats.put("Token Count", results);
+
+        results = getLayerStatistics(aStatisticRequest, "<Token=\"\"/>", fullDocSet, true);
+        results.remove("stats");
+        results.remove("sourceNumber");
+        results.put("Number of Documents", results.get("n"));
+        results.remove("n");
+        allStats.put("per Sentence: Token Count", results);
+        nonNullStats.put("per Sentence: Token Count", results);
+
+        results = getLayerStatistics(aStatisticRequest, "<s=\"\"/>", fullDocSet, false);
+        results.remove("stats");
+        results.remove("sourceNumber");
+        results.put("Number of Documents", results.get("n"));
+        results.remove("n");
+        allStats.put("Sentence Count", results);
+        nonNullStats.put("Sentence Count", results);
+
+        return new StatisticsResult(aStatisticRequest, allStats, nonNullStats);
+    }
+
+    @Override
+    public ArrayList<Integer> getUniqueDocuments(StatisticRequest aStatisticRequest)
+        throws IOException
+    {
+        IndexSearcher searcher = null;
+        Map<Long, Long> annotatableDocuments = listAnnotatableDocuments(
```

Comment:
What is this mapping and why is it Long, Long?

Reply:
listAnnotatableDocuments has been there before (MtasDocumentIndex around line 800). I assume that the document IDs can become so large that int will not always suffice. 
The keys of annotatableDocuments are the IDs of source documents which have a annotation document linked to them. The ID of the annotation document is the corresponding value of this key.